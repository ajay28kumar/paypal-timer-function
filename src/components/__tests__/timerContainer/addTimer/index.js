import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AddTimer, {
  isValidStartTime,
  createUUID,
} from "../../../timerContainer/addTimer";

const mockSetTimerDetails = jest.fn();

const props = {
  timerDetails: [{ id: "1236", startTime: 56 }],
  setTimerDetails: mockSetTimerDetails,
};

describe("addTimerComponent component tests", () => {
  it("snapshot testing with props", () => {
    const app = shallow(<AddTimer {...props} />);
    expect(toJson(app)).toMatchSnapshot();
  });
  describe("test isValidStartTime", () => {
    it("return true for integer 5", () => {
      expect(isValidStartTime(5)).toBeTruthy();
    });
    it("return false for negative values", () => {
      expect(isValidStartTime(-8)).toBeFalsy();
    });
  });
  describe("test uuid", () => {
    it("return 36 random values", () => {
      expect(createUUID().length).toBe(36);
    });
  });
  describe("addtimer function test", function () {
    it("button click invoke setTimerDetails", () => {
      const app = shallow(<AddTimer {...props} />);
      const button = app.find("WithStyles(ForwardRef(Button))");
      button.simulate("click");
      expect(mockSetTimerDetails).toHaveBeenCalled();
    });
  });
});
