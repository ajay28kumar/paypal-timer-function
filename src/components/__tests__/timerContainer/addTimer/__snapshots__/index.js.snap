// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addTimerComponent component tests snapshot testing with props 1`] = `
<div
  className="makeStyles-timerContainer-1"
>
  <div
    className="makeStyles-addTimeInputContainer-2"
  >
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-inputBox-3"
      fullWidth={true}
      label="Enter Time in Seconds"
      onChange={[Function]}
      placeholder="Enter Countdown Timer"
      value=""
    />
    <WithStyles(ForwardRef(Button))
      color="primary"
      disabled={false}
      onClick={[Function]}
      variant="outlined"
    >
      Add Timer
    </WithStyles(ForwardRef(Button))>
  </div>
</div>
`;
